zorg/buildbot/builders/ClangBuilder.py

```diff
@@ -427,9 +436,31 @@ def _getClangCMakeBuildFactory(
         # Note: Backslash path separators do not work well with cmake and ninja.
         # Forward slash path separator works on Windows as well.
         stage1_cc = InterpolateToPosixPath(
-                        f"-DCMAKE_C_COMPILER=%(prop:builddir)s/{stage1_install}/bin/{cc}")
+                        f"%(prop:builddir)s/{stage1_install}/bin/{cc}")
         stage1_cxx = InterpolateToPosixPath(
-                        f"-DCMAKE_CXX_COMPILER=%(prop:builddir)s/{stage1_install}/bin/{cxx}")
+                        f"%(prop:builddir)s/{stage1_install}/bin/{cxx}")
+
+        # If stage2_toolchain_options is set when we'll use a toolchain file
+        # to specify the compiler being used (the just-built stage1) and add
+        # any stage2_toolchain_options to it. Otherwise, just set
+        # -DCMAKE_{C,CXX}_COMPILER.
+        if stage2_toolchain_options is None:
+            compiler_args = [
+                    f"-DCMAKE_C_COMPILER={stage1_cc}",
+                    f"-DCMAKE_CXX_COMPILER={stage1_cxx}"
```

Comment:
I thought maybe we should use a toolchain file for all 2 stage builds but:
* It's not recommended practice if you aren't cross compiling.
* CMAKE_C_COMPILER and friends would not be visible in the log files for stage 2.
* Reproducing a build now has the extra step of writing the toolchain file, even if it was just copy paste.

So I agree with the decision to not change how the existing builds work.

If transparency were our only goal here, I'd say to pass all the toolchain file contents as `-D...` but you said:
> (notably, some CMake variables can't be set outside of a toolchain file)

Which answers that (and I have had similar experiences with CMake).

Reply:
Crazy idea though, could the toolchain file's contents be pulled from a file here in the llvm-zorg repo?

It could be like:
```
<the options you set in the builder already>
# Set your host compiler here.
# set(CMAKE_C_COMPILER ....)
# set(CMAKE_CXX_COMPILER ...)
```

Then when someone wants to reproduce, there is a starting point that's already in the right form to use.

Reply:
In not actual Python terms:
```
stage2_toolchain_options=[open(file).splitlines()]
```